resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/SparkKubernetesClientFactory.scala

```diff
@@ -67,8 +66,16 @@ private[spark] object SparkKubernetesClientFactory {
     val dispatcher = new Dispatcher(
       ThreadUtils.newDaemonCachedThreadPool("kubernetes-dispatcher"))
 
-    // TODO [SPARK-25887] Create builder in a way that respects configurable context
-    val config = new ConfigBuilder()
+    // Allow for specifying a context used to auto-configure from the users K8S config file
+    val kubeContext = sparkConf.get(KUBERNETES_CONTEXT).filter(_.nonEmpty)
+    logInfo(s"Auto-configuring K8S client using " +
+      s"${if (kubeContext.isDefined) s"context ${kubeContext.getOrElse("?")}" else "current context"}" +
```

Comment:
I saw your commit about the NPE, but I'm not sure how you could get one here?

`sparkConf.get` will never return `Some(null)` as far as I know (it would return `None` instead).

Reply:
I think using `kubeContext.map("context " + _).getOrElse("current context")` would make this cleaner.